js/activityDetection/activityDetection.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const systemIdleTime = require('@paulcbetts/system-idle-time');
+const throttle = require('../utils/throttle');
+
+const activity = require('./activity.js');
+const maxIdleTime = 4 * 60 * 1000;
+let intervalId;
+
+/**
+ * @return {{isUserIdle: boolean, systemIdleTime: *}}
+ * Check if the user is idle
+ */
+function activityDetection() {
+    // Get system idle status and idle time from PaulCBetts package
+    if (systemIdleTime.getIdleTime() < maxIdleTime) {
+        return {isUserIdle: false, systemIdleTime: systemIdleTime.getIdleTime()};
+    }
+
+    // If idle for more than 4 mins, monitor system idle status every second
+    if (!intervalId) monitorUserActivity();
+}
+
+/**
+ * Start monitoring user activity status.
+ * Run every 4 mins to check user idle status
+ */
+function initiateActivityDetection() {
+    let activityCheckInterval = 4 * 60 * 1000;
+
+    let throttleActivity = throttle(activityCheckInterval, sendActivity);
+    setInterval(throttleActivity, 5000);
```

Comment:
should make sure to cancel old timer if initiateActivityDetection is called again.  see comment below about reload.